src/phoenix/trace/llama_index/callback.py

```diff
@@ -339,29 +343,35 @@ def _add_spans_to_tracer(
     while parent_child_id_stack:
         parent_span_id, event_id = parent_child_id_stack.pop()
         event_data = event_id_to_event_data[event_id]
-        event_type = event_data["event_type"]
-        attributes = event_data["attributes"]
+        event_type = event_data.event_type
+        attributes = event_data.attributes
         if event_type is CBEventType.LLM:
             while parent_child_id_stack:
                 preceding_event_parent_span_id, preceding_event_id = parent_child_id_stack[-1]
                 if preceding_event_parent_span_id != parent_span_id:
                     break
                 preceding_event_data = event_id_to_event_data[preceding_event_id]
-                if preceding_event_data["event_type"] is not CBEventType.TEMPLATING:
+                if preceding_event_data.event_type is not CBEventType.TEMPLATING:
                     break
                 parent_child_id_stack.pop()
-                if payload := preceding_event_data["start_event"].payload:
-                    # Add template attributes to the LLM span to which they belong.
-                    attributes.update(_template_attributes(payload))
+                if preceding_event_start := preceding_event_data.start_event:
+                    if preceding_payload := preceding_event_start.payload:
```

Comment:
nit. since it's always a conjunction, we should make the intention clear by writing it as a conjunction
```suggestion
                if preceding_event_start := preceding_event_data.start_event and preceding_payload := preceding_event_start.payload:
```